### DISCUSSION TITLE
How to run the toolchain binaries use `bazel run @bazel_embedded//toolchains/.....`

### DISCUSSION THREAD

**Discussion by cbrune** [bazelembedded/bazel-embedded#54] [Q&A] [answered]
I've built a binary using bazel_embedded.  It works great, thanks for all your work.

Now, however, I want to run `objdump` on the result to inspect it.  What is the best way to invoke `objdump` using the same toolchain that bazel_embedded uses?  Or by extension, any of the binaries from the toolchain?

I want to avoid installing any toolchain in my host environment.  All the tools should come from the bazel cache.

**Comment by nathaniel-brough** [comment, marked as answer]
All the tools downloaded as part of gcc-arm-none are exported and available under the target;

@com_gcc_arm_none_eabi_compiler//:bin/arm-none-eabi-*

So you could put together a quick prototype to build something using objdump like so (untested);
```py
genrule(
    name = "my_elf_symbols",
    srcs = [":my_elf"],
    outs = ["my_elf_symbols.txt"],
    cmd = "./$(location @com_gcc_arm_none_eabi_compiler//:bin/arm-none-eabi-objdump) -TC $(location :my_elf) > $@",
    tools = ["@com_gcc_arm_none_eabi_compiler//:bin/arm-none-eabi-objdump"],
)
```

If you are after something more sophisticated you might consider building some custom rules.

You can also run those binaries directly e.g.

`bazel run @com_gcc_arm_none_eabi_compiler//:bin/arm-none-eabi-objdump -- -TC bazel-bin/path/to/elf`

**Reply by cbrune** [reply]
Wonderful!  Thank you.  The interactive `bazel run @com_gcc_arm_none_eabi_compiler//:bin/arm-none-eabi-objdump ....` was what I was looking for.  Thanks again.
